pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -1182,11 +1181,13 @@ func (cb *ClusterBuilder) buildUpstreamClusterTLSContext(opts *buildClusterOpts,
 	return tlsContext, nil
 }
 
-// ApplyECDHCurves applies the ECDH curves from mesh config to UpstreamTlsContext
-// Used for building upstream TLS context for mesh external TLS/mTLS origination
-func ApplyECDHCurves(tlsContext *auth.UpstreamTlsContext, mesh *meshconfig.MeshConfig) {
-	if mesh != nil && mesh.TlsDefaults != nil && len(mesh.TlsDefaults.EcdhCurves) > 0 {
-		tlsContext.CommonTlsContext.TlsParams.EcdhCurves = mesh.TlsDefaults.EcdhCurves
+// ApplyTLSDefaults applies tls default settings from mesh config to UpstreamTlsContext.
+func ApplyTLSDefaults(tlsContext *auth.UpstreamTlsContext, tlsDefaults *meshconfig.MeshConfig_TLSConfig) {
+	if tlsDefaults == nil {
+		return
+	}
+	if len(tlsDefaults.EcdhCurves) > 0 {
```

Comment:
do we plan to use tlsVersion for egress as well? I understand that currently it is hardcoded.

Reply:
Since minTLSVersion in mTLSSettings is applied only for server side - keeping this same as that. Typically it is controlled/dictated at the server